org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.java

```diff
@@ -232,18 +231,25 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
 		result.setCapabilities(createServerCapabilities(params));
 		access.addBuildListener(this);
 		return requestManager.runWrite(() -> {
-			if (workspaceManager.isSupportsWorkspaceFolders()) {
+			if (clientSupportsWorkspaceFolders() && workspaceManager.isSupportsWorkspaceFolders()) {
 				List<WorkspaceFolder> workspaceFolders = params.getWorkspaceFolders();
 				if (workspaceFolders == null)
 					workspaceFolders = Collections.emptyList();
 				workspaceManager.initialize(workspaceFolders, this::publishDiagnostics, CancelIndicator.NullImpl);
 			} else {
+				URI baseDir = getBaseDir(params);
 				workspaceManager.initialize(baseDir, this::publishDiagnostics, CancelIndicator.NullImpl);
 			}
 			return result;
 		}, (cancelIndicator, it) -> it).thenApply(it -> initializeResult = it);
 	}
 
+	protected boolean clientSupportsWorkspaceFolders() {
+		return this.initializeParams.getCapabilities() != null 
+				&& this.initializeParams.getCapabilities().getWorkspace() != null 
+				&& this.initializeParams.getCapabilities().getWorkspace().getWorkspaceFolders() == Boolean.TRUE;
```

Comment:
Because paranoia, I'd use `Objects.equals(this.initializeParams.getCapabilities().getWorkspace().getWorkspaceFolders(), Boolean.TRUE)`

Reply:
I also removed check on getCapabilities as it is notNull annotated

Reply:
Somehow it gets null sometimes, so I added the check again.